File: epi_judge_python/parity.py
def parity(x):
    # Implement this placeholder.
    parity = 0
    while x:
        parity ^= 1
        x &= x - 1
    return parity


from test_framework import test_utils_generic_main, test_utils

if __name__ == '__main__':
    test_utils_generic_main.generic_test_main('parity.tsv', parity)

File: epi_judge_cpp/parity.cc
short Parity(unsigned long x) {
  // Implement this placeholder.
  short parity = 0;
  while ( x > 0 ) {
	  parity ^= 1;
	  x = x & x - 1;
	}
  return parity;
}

#include "test_framework/test_utils_generic_main.h"

int main(int argc, char* argv[]) {
  std::vector<std::string> param_names{"x"};
  generic_test_main(argc, argv, param_names, "parity.tsv", &Parity);
  return 0;
}
